drivers/ieee802154/ieee802154_cc13xx_cc26xx_subg.c

```diff
@@ -419,20 +453,17 @@ static int ieee802154_cc13xx_cc26xx_subg_set_channel(
 	drv_data->cmd_fs.status = IDLE;
 	drv_data->cmd_fs.frequency = freq;
 	drv_data->cmd_fs.fractFreq = fract;
-	reason = RF_runCmd(drv_data->rf_handle, (RF_Op *)&drv_data->cmd_fs,
+	events = RF_runCmd(drv_data->rf_handle, (RF_Op *)&drv_data->cmd_fs,
 			   RF_PriorityNormal, NULL, 0);
-	if (reason != RF_EventLastCmdDone) {
-		LOG_DBG("Failed to set frequency: 0x%" PRIx64, reason);
-		r = -EIO;
+	if (events != RF_EventLastCmdDone) {
+		LOG_DBG("Failed to set frequency: 0x%" PRIx64, events);
+		ret = -EIO;
 		goto out;
 	}
 
-	/* Run BG receive process on requested channel */
```

Comment:
I'm not 100% sure it's safe to remove this

Reply:
In any case leaving RX on after setting the channel would be a bug as most of the time the radio is off when you set the channel and then leaving RX on would drain the battery. That's why I removed it.

Setting the channel must be done while not associated and scanning deals with switching the device on (aka going into RX mode) - so in practice it shouldn't be a problem.

But I agree that it would be safer to check whether RX is on at the beginning of the CCA routine (looking at the cmd's status == 2) and switch it back on if it was on before and leave it off otherwise. I'll fix that.

Reply:
@cfriedt Can you have a look again? I think I fixed this now.